transport/http.go

```diff
@@ -363,6 +375,9 @@ func (t *HTTPTransport) SendStream(ctx context.Context, r io.Reader) error {
 	req.Body = ioutil.NopCloser(r)
 	if err := t.sendStreamRequest(req); err != nil {
 		atomic.StoreInt32(&t.urlIndex, (urlIndex+1)%int32(len(t.intakeURLs)))
+		// The remote APM Server url has changed, so we refresh the remote APM
+		// Server version for the new URL.
+		t.refreshRemoteVersion(ctx)
```

Comment:
I don't have a strong reason for this at the moment, but I think it might be neater to only every query the version from MajorServerVersion. Here we could just invalidate the cache, causing the next call to MajorServerVersion (if there is one) to refresh it. WDYT?